native/src/main/java/io/ballerina/stdlib/io/nativeimpl/CsvChannelUtils.java

```diff
@@ -60,23 +61,31 @@ public class CsvChannelUtils {
     private static final BString ENCODING = StringUtils.fromString("UTF-8");
 
     public static Object fileReadCsv(BString path, int skipHeaders, BTypedesc typeDesc) {
-        BObject byteChannel = (BObject) ByteChannelUtils.openReadableFile(path);
+        Object byteChannelObject = ByteChannelUtils.openReadableFile(path);
+        if (byteChannelObject instanceof BError) {
+            return IOUtils.createError(IOConstants.ErrorCode.FileNotFoundError,
+                    ((BError) byteChannelObject).getMessage());
+        }
+        BObject byteChannel = (BObject) byteChannelObject;
         BObject characterChannel = ValueCreator.createObjectValue(getIOPackage(),
             READABLE_CHARACTER_CHANNEL, byteChannel, ENCODING);
         BObject textRecordChannel = ValueCreator.createObjectValue(getIOPackage(),
             READABLE_TEXT_RECORD_CHANNEL, characterChannel, FIELD_SEPERATOR, ROW_SEPERATOR, FORMAT);
         textRecordChannel.addNativeData(CSV_RETURN_TYPE, typeDesc);
         while (hasNext(textRecordChannel)) {
-
             return getAllRecords(textRecordChannel, skipHeaders, typeDesc);
         }
         return null;
     }
 
-    public static BStream createCsvAsStream(BString path, BTypedesc typeDesc) {
+    public static Object createCsvAsStream(BString path, BTypedesc typeDesc) {
```

Comment:
Will this change results in the API returning stream<>|Error?

Reply:
Yes. we have to use Objects as the return type cannot be changed in the case for BObject.

Reply:
In the ballerina side, the external function returns the `stream<returnType, Error?>|Error` by default. So there is no requirement to change the output type.